src/operator/image/image_random-inl.h

```diff
@@ -62,6 +67,23 @@ inline bool ToTensorType(const nnvm::NodeAttrs& attrs,
   return (*in_attrs)[0] != -1;
 }
 
+void ToTensorImpl(const std::vector<TBlob> &inputs,
+                        const std::vector<TBlob> &outputs,
+                        const int length,
+                        const int channel,
+                        const int step = 0) {
+  MSHADOW_TYPE_SWITCH(inputs[0].type_flag_, DType, {
+      float* output = outputs[0].dptr<float>();
+      DType* input = inputs[0].dptr<DType>();
+
```

Comment:
```suggestion
#pragma omp parallel for collapse(2)
```
would this make better performance?

Reply:
Cannot be used inside Macro.

Reply:
How about checking the DType by ourself without using the Macro